docs/source/format/CanonicalExtensions.rst

```diff
@@ -417,7 +421,584 @@ better zero-copy compatibility with various systems that also store booleans usi
+a lossless fashion. This also provides the ability to represent `shredded <https://github.com/apache/parquet-format/blob/master/VariantShredding.md>`__
+variant values. This will make it possible for systems to pass Variant data around without having to upgrade their Arrow version
+or otherwise require special handling unless they want to directly interact with the encoded variant data. See the previous links
+to the Parquet format specification for details on what the actual binary values should look like.
+
+* Extension name: ``parquet.variant``.
+
+* The storage type of this extension is a ``Struct`` that obeys the following rules:
+
+  * A *non-nullable* field named ``metadata`` which is of type ``Binary``, ``LargeBinary``, or ``BinaryView``.
+
+  * At least one (or both) of the following:
+
+    * A field named ``value`` which is of type ``Binary``, ``LargeBinary``, or ``BinaryView``.
+      *(unshredded variants consist of just the ``metadata`` and ``value`` fields only)*
+
+    * A field named ``typed_value`` which can be any :term:`primitive type` or a ``List``, ``LargeList``, ``ListView`` or ``Struct``
+
+      * If the ``typed_value`` field is a *nested* type, its elements **must** be *non-nullable* and **must** be a ``Struct`` consisting of
+        at least one (or both) of the following:
+
+        * A field named ``value`` which is of type ``Binary``, ``LargeBinary``, or ``BinaryView``.
+
+        * A field named ``typed_value`` which follows these same rules for the ``typed_value`` field.
+
+.. note::
+
+   It is also *permissible* for the ``metadata`` field to be dictionary-encoded with an index type of ``int8``.
+
+.. note::
+
+   The fields may be in any order, and thus must be accessed by **name** not by *position*. The field names are case sensitive.
+
+Examples
+--------
+
+Unshredded
+''''''''''
+
+The simplest case, an unshredded variant always consists of **exactly** two fields: ``metadata`` and ``value``. Any of
+the following storage types are valid (not an exhaustive list):
+
+* ``struct<metadata: binary non-nullable, value: binary nullable>``
+* ``struct<value: binary nullable, metadata: binary non-nullable>``
+* ``struct<metadata: dictionary<int8, binary> non-nullable, value: binary_view nullable>``
+
+Simple Shredding
+''''''''''''''''
+
+Suppose we have a Variant field named *measurement* and we want to shred the ``int64`` values into a separate column for efficiency.
+In Parquet, this could be represented as::
+
+  required group measurement (VARIANT) {
+    required binary metadata;
+    optional binary value;
+    optional int64 typed_value;
+  }
+
+Thus the corresponding storage type for the ``parquet.variant`` Arrow extension type would be::
+
+  struct<
+    metadata: binary required,
+    value: binary optional,
+    typed_value: int64 optional
+  >
+
+If we suppose a series of measurements consisting of::
```

Comment:
The concrete example doesn't seem required to me? @wgtmac What do you think?

Reply:
I'm not sure if examples of data layout are required because these are not new knowledge to readers. However, it might be useful for developers who implement the Variant type support to get started with these examples?

Reply:
As I said above, my vote would be now on a separate document.

Reply:
moved the examples to a separate document